### PR TITLE
Extract NaCl-specific entry point in C++ Example

### PR DESCRIPTION
**Pull request by emaxx-google**
Split the pp_module.cc source file in //example_cpp_smart_card_client_app
into two parts: NaCl-specific "entry_point_nacl.cc" and
toolchain-independent "application.{h|cc}".

This is a preparatory CL for implementing a different entry point - the
one for Emscripten/WebAssembly builds (as tracked by #220).

### CODE REVIEW

**Comment by emaxx-google** [comment]
PTAL. The CL is big, but it's mostly just moving the code around. No functional changes were made; some of comments were edited to get rid of NaCl-specific details in toolchain.{h|cc}.

File: example_cpp_smart_card_client_app/src/application.h [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+// dialog integration.
+//
+// As an example, this implementation starts a background thread running the
+// Work method after the initialization happens.
+//
+// Please note that all blocking operations (for example, PC/SC-Lite API calls
+// or PIN requests) should be never executed on the main event loop thread. This
+// is because all communication with the JavaScript side works through
+// exchanging of messages between the executable module and JavaScript side, and
+// the incoming messages are received and routerd on the main thread (see
```
Comment (Fabian-Sommer) [inline example_cpp_smart_card_client_app/src/application.h:40]: nit: typo in routed

File: example_cpp_smart_card_client_app/src/entry_point_nacl.cc [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+// The implementation presented here is a wrapper around the `Application` class
+// that is a toolchain-independent skeleton (it can be compiled under both
+// Emscripten/WebAssembly and Native Client).
+class PpInstance final : public pp::Instance {
+ public:
+  // The constructor that is executed during the NaCl module startup.
+  //
+  // The implementation presented here does the following:
+  // * creates a `google_smart_card::GlobalContextImplNacl` class instance that
+  //   is used by the Application class to perform operations that has to be
```
Comment (Fabian-Sommer) [inline example_cpp_smart_card_client_app/src/entry_point_nacl.cc:57]: nit: "have" instead of "has"

**Review by Fabian-Sommer — APPROVED** [review]
LGTM % nits

**Comment by emaxx-google** [comment]
Thank you.

**Comment by emaxx-google** [comment]
Split the pp_module.cc source file in //example_cpp_smart_card_client_app
into two parts: NaCl-specific "entry_point_nacl.cc" and
toolchain-independent "application.{h|cc}".

This is a preparatory CL for implementing a different entry point - the
one for Emscripten/WebAssembly builds (as tracked by #220).

**Pull request merged**